File: benchmark/python/fib_iterative.py
import time

def fib(n):
    if n <= 1:
        return n
    a = 0
    b = 1
    for i in range(2, n + 1):
        temp = a + b
        a = b
        b = temp
    return b

start = time.time() * 1000
s = 0
for i in range(10000):
    s = s + fib(1000)
end = time.time() * 1000
print(f"Sum: {s}")
print(f"Time: {end - start:.0f}ms")

File: benchmark/ruby/fib_iterative.rb
start = Time.now.to_f * 1000
def fib(n)
    return n if n <= 1
    a = 0
    b = 1
    (2..n).each do |i|
        temp = a + b
        a = b
        b = temp
    end
    b
end
s = 0
10000.times do
    s = s + fib(1000)
end
finish = Time.now.to_f * 1000
puts "Sum: #{s}"
puts "Time: #{(finish - start).round}ms"

